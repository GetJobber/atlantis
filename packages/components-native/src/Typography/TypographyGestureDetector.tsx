import type { PropsWithChildren } from "react";
import React from "react";
import { Gesture, GestureDetector } from "react-native-gesture-handler";

/**
 * This is a workaround suggested by react-native-gesture-handler to prevent
 * accidental highlighting of text in Android devices
 * https://github.com/software-mansion/react-native-gesture-handler/issues/1372
 */
export function TypographyGestureDetector(
  props: PropsWithChildren<unknown>,
): JSX.Element {
  return <GestureDetector {...props} gesture={Gesture.Native()} />;
}
