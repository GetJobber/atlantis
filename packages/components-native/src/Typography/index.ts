export { Typography } from "./Typography";
export type {
  FontFamily,
  FontStyle,
  FontWeight,
  BaseWeight,
  DisplayWeight,
  BaseStyle,
  DisplayStyle,
  TextColor,
  TextTransform,
  TextSize,
  TextAlign,
  LineHeight,
  LetterSpacing,
  TextAccessibilityRole,
  TextVariation,
  TypographyProps,
  TruncateLength,
} from "./Typography";
export { typographyStyles } from "./Typography.style";
export { TypographyGestureDetector } from "./TypographyGestureDetector";
