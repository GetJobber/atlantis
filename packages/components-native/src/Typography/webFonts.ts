import type { TextStyle } from "react-native";

export const webFonts: { [index: string]: TextStyle } = {
  baseRegularRegular: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "400",
  },

  baseRegularMedium: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "500",
  },

  baseRegularSemiBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "600",
  },

  baseRegularBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "700",
  },

  baseRegularExtraBold: {
    fontFamily: "'Inter', Helvetica, Arial, sans-serif",
    fontWeight: "800",
  },

  displayRegularBold: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "700",
  },

  displayRegularExtraBold: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "800",
  },

  displayRegularBlack: {
    fontFamily: "'Jobber Pro', 'Poppins', Helvetica, Arial, sans-serif",
    fontWeight: "900",
  },
};
