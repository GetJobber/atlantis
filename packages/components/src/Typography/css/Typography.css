@import url("https://fonts.googleapis.com/css?family=Poppins:800,900|Source+Sans+Pro:400,400i,700,700i&display=swap");

:root {
  --typography--fontFamily-normal: "Source Sans Pro", Helvetica, Arial,
    sans-serif;
  --typography--fontFamily-display: "Poppins", Helvetica, Arial, sans-serif;

  --typography--fontSize-extravagant: calc(var(--base-unit) * 3); /* 48 */
  --typography--fontSize-jumbo: calc(var(--base-unit) * 2.25); /* 36 */
  --typography--fontSize-largest: calc(var(--base-unit) * 1.5); /* 24 */
  --typography--fontSize-larger: calc(var(--base-unit) * 1.25); /* 20 */
  --typography--fontSize-large: calc(var(--base-unit) * 1.125); /* 18 */
  --typography--fontSize-base: calc(var(--base-unit) * 1); /* 16 */
  --typography--fontSize-small: calc(var(--base-unit) * 0.875); /* 14 */
  --typography--fontSize-smaller: calc(var(--base-unit) * 0.75); /* 12 */

  --typography--lineHeight-large: 1.34;
  --typography--lineHeight-base: 1.25;
  --typography--lineHeight-tight: 1.2;
  --typography--lineHeight-tighter: 1.143;
  --typography--lineHeight-tightest: 1.12;
  --typography--lineHeight-miniscule: 1.08;
}

.base {
  margin: 0;
  padding: 0;
  color: var(--color-blue);
  font-family: var(--typography--fontFamily-normal);
  font-size: var(--typography--fontSize-base);
  font-weight: 400;
  line-height: var(--typography--lineHeight-base);
}
