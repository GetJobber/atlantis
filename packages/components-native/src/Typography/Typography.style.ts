import { Platform, StyleSheet, TextStyle } from "react-native";
import { webFonts } from "./webFonts";
import { tokens } from "../utils/design";

const extravagantLineHeight = tokens["typography--lineHeight-extravagant"];
const jumboLineHeight = tokens["typography--lineHeight-jumbo"];
const largestLineHeight = tokens["typography--lineHeight-largest"];
const largerLineHeight = tokens["typography--lineHeight-larger"];
const largeLineHeight = tokens["typography--lineHeight-large"];
const baseLineHeight = tokens["typography--lineHeight-base"];
const tightLineHeight = tokens["typography--lineHeight-tight"];
const minisculeLineHeight = tokens["typography--lineHeight-miniscule"];

const deviceFonts = {
  baseRegularRegular: {
    fontFamily: "inter-regular",
  },

  baseRegularMedium: {
    fontFamily: "inter-medium",
  },

  baseRegularBold: {
    fontFamily: "inter-bold",
  },

  baseRegularSemiBold: {
    fontFamily: "inter-semibold",
  },

  baseRegularExtraBold: {
    fontFamily: "inter-extrabold",
  },

  baseItalicRegular: {
    fontFamily: "inter-italic",
  },

  baseItalicMedium: {
    fontFamily: "inter-italic-medium",
  },

  baseItalicBold: {
    fontFamily: "inter-italic-bold",
  },

  baseItalicSemiBold: {
    fontFamily: "inter-italic-semibold",
  },

  baseItalicExtraBold: {
    fontFamily: "inter-italic-extrabold",
  },

  displayRegularBold: {
    fontFamily: "jobberpro-bd",
  },

  displayRegularExtraBold: {
    fontFamily: "jobberpro-xbd",
  },

  displayRegularBlack: {
    fontFamily: "jobberpro-blk",
  },
};

/**
 * We need to use web fonts for rendering Typography on Storybook
 * because it uses font files (.ttf) to render them on devices.
 * As we don't want to expose the font files, we are setting the fonts
 * in CSS.
 */
const fonts = Platform.select({
  web: webFonts,
  default: deviceFonts,
});

/**
 * Reusable typography tokens to ensure consistency for any client facing texts.
 */
export const typographyTokens: { [index: string]: TextStyle } = {
  // This follows a pattern of
  // { fontFamily }{ fontStyle }{ fontWeight }
  ...fonts,

  italic: {
    fontStyle: "italic",
  },

  startAlign: {
    textAlign: "left",
  },

  endAlign: {
    textAlign: "right",
  },

  centerAlign: {
    textAlign: "center",
  },

  justifyAlign: {
    textAlign: "justify",
  },

  blue: {
    color: tokens["color-heading"],
  },

  blueDark: {
    color: tokens["color-blue--dark"],
  },

  white: {
    color: tokens["color-white"],
  },

  green: {
    color: tokens["color-green"],
  },

  greenDark: {
    color: tokens["color-green--dark"],
  },

  grey: {
    color: tokens["color-grey"],
  },

  greyDark: {
    color: tokens["color-grey--dark"],
  },

  greyBlue: {
    color: tokens["color-greyBlue"],
  },

  greyBlueDark: {
    color: tokens["color-greyBlue--dark"],
  },

  lightBlue: {
    color: tokens["color-lightBlue"],
  },

  lightBlueDark: {
    color: tokens["color-lightBlue--dark"],
  },

  red: {
    color: tokens["color-red"],
  },

  redDark: {
    color: tokens["color-red--dark"],
  },

  yellow: {
    color: tokens["color-yellow"],
  },

  yellowDark: {
    color: tokens["color-yellow--dark"],
  },

  yellowGreenDark: {
    color: tokens["color-yellowGreen--dark"],
  },

  orangeDark: {
    color: tokens["color-orange--dark"],
  },

  navyDark: {
    color: tokens["color-navy--dark"],
  },

  limeDark: {
    color: tokens["color-lime--dark"],
  },

  purpleDark: {
    color: tokens["color-purple--dark"],
  },

  pinkDark: {
    color: tokens["color-pink--dark"],
  },

  tealDark: {
    color: tokens["color-teal--dark"],
  },

  indigoDark: {
    color: tokens["color-indigo--dark"],
  },

  navy: {
    color: tokens["color-navy"],
  },

  heading: {
    color: tokens["color-heading"],
  },

  headingReverse: {
    color: tokens["color-text--reverse"],
  },

  text: {
    color: tokens["color-text"],
  },

  textSecondary: {
    color: tokens["color-text--secondary"],
  },

  textReverse: {
    color: tokens["color-text--reverse"],
  },

  textReverseSecondary: {
    color: tokens["color-text--reverse--secondary"],
  },

  success: {
    color: tokens["color-success--onSurface"],
  },

  error: {
    color: tokens["color-critical"],
  },

  base: {
    color: tokens["color-text"],
  },

  subdued: {
    color: tokens["color-text--secondary"],
  },

  warn: {
    color: tokens["color-warning--onSurface"],
  },

  info: {
    color: tokens["color-informative--onSurface"],
  },

  critical: {
    color: tokens["color-critical"],
  },

  successReverse: {
    color: tokens["color-success"],
  },

  errorReverse: {
    color: tokens["color-critical"],
  },

  baseReverse: {
    color: tokens["color-text--reverse"],
  },

  subduedReverse: {
    color: tokens["color-text--reverse--secondary"],
  },

  warnReverse: {
    color: tokens["color-warning"],
  },

  infoReverse: {
    color: tokens["color-informative"],
  },

  criticalReverse: {
    color: tokens["color-critical"],
  },

  interactive: {
    color: tokens["color-interactive"],
  },

  destructive: {
    color: tokens["color-destructive"],
  },

  learning: {
    color: tokens["color-informative"],
  },

  subtle: {
    color: tokens["color-interactive--subtle"],
  },

  onPrimary: {
    color: tokens["color-surface"],
  },

  disabled: {
    color: tokens["color-disabled"],
  },

  smallestSize: {
    fontSize: tokens["typography--fontSize-smallest"],
    lineHeight: minisculeLineHeight,
  },

  smallerSize: {
    fontSize: tokens["typography--fontSize-smaller"],
    lineHeight: tightLineHeight,
  },

  smallSize: {
    fontSize: tokens["typography--fontSize-small"],
    lineHeight: tightLineHeight,
  },

  defaultSize: {
    fontSize: tokens["typography--fontSize-base"],
    lineHeight: baseLineHeight,
  },

  largeSize: {
    fontSize: tokens["typography--fontSize-large"],
    lineHeight: largeLineHeight,
  },

  largerSize: {
    fontSize: tokens["typography--fontSize-larger"],
    lineHeight: largeLineHeight,
  },

  largestSize: {
    fontSize: tokens["typography--fontSize-largest"],
    lineHeight: largerLineHeight,
  },

  jumboSize: {
    fontSize: tokens["typography--fontSize-jumbo"],
    lineHeight: jumboLineHeight,
  },

  extravagantSize: {
    fontSize: tokens["typography--fontSize-extravagant"],
    lineHeight: extravagantLineHeight,
  },

  extravagantLineHeight: {
    lineHeight: extravagantLineHeight,
  },

  jumboLineHeight: {
    lineHeight: jumboLineHeight,
  },

  largestLineHeight: {
    lineHeight: largestLineHeight,
  },
  largerLineHeight: {
    lineHeight: largerLineHeight,
  },

  largeLineHeight: {
    lineHeight: largeLineHeight,
  },

  baseLineHeight: {
    lineHeight: baseLineHeight,
  },

  tightLineHeight: {
    lineHeight: tightLineHeight,
  },

  baseLetterSpacing: {
    letterSpacing: tokens["typography--letterSpacing-base"],
  },

  looseLetterSpacing: {
    letterSpacing: tokens["typography--letterSpacing-loose"],
  },

  strikeThrough: {
    textDecorationLine: "line-through",
  },

  underline: {
    textDecorationLine: "underline",
  },

  underlineAndStrikeThrough: {
    textDecorationLine: "underline line-through",
  },
};

/**
 * `StyleSheet` for Typography.tsx.
 *
 * If you find yourself needing to use what's inside this object on files other
 * than `<Typography />`, please import from `@jobber/components-native` instead.
 *
 * ```
 * import { typographyStyles } from "@jobber/components-native"
 * ```
 */
export const typographyStyles: { [index: string]: TextStyle } =
  StyleSheet.create(typographyTokens);
