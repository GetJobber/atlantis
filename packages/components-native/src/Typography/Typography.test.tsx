import React from "react";
import { render } from "@testing-library/react-native";
import { I18nManager } from "react-native";
import { Typography } from "./Typography";

it("renders text with no additional props", () => {
  const typography = render(<Typography>Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with regular style", () => {
  const typography = render(
    <Typography fontStyle="regular">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with bold style", () => {
  const typography = render(
    <Typography fontWeight="bold">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with italic style", () => {
  const typography = render(
    <Typography fontStyle="italic">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with bold weight and italic style", () => {
  const typography = render(
    <Typography fontStyle="italic" fontWeight="bold">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with bold style and display as fontFamily", () => {
  const typography = render(
    <Typography fontFamily="display" fontWeight="bold">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with extraBold weight and display as fontFamily", () => {
  const typography = render(
    <Typography fontFamily="display" fontWeight="extraBold">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with black style and display as fontFamily", () => {
  const typography = render(
    <Typography fontFamily="display" fontWeight="black">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with uppercase transform", () => {
  const typography = render(
    <Typography transform="uppercase">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with lowercase transform", () => {
  const typography = render(
    <Typography transform="lowercase">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with white color", () => {
  const typography = render(<Typography color="white">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with green color", () => {
  const typography = render(<Typography color="green">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with default color", () => {
  const typography = render(<Typography color="default">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with center align", () => {
  const typography = render(<Typography align="center">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text respecting the text direction", () => {
  I18nManager.isRTL = true;

  const typography = render(<Typography>Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
  I18nManager.isRTL = false;
});

it("renders text with small size", () => {
  const typography = render(<Typography size="small">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with large size", () => {
  const typography = render(<Typography size="large">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with default size", () => {
  const typography = render(<Typography size="default">Test Text</Typography>);
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with multiple properties", () => {
  const typography = render(
    <Typography fontWeight="bold" size="large" color="white">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with line height override", () => {
  const typography = render(
    <Typography lineHeight="jumbo">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with letter spacing", () => {
  const typography = render(
    <Typography letterSpacing="loose">Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with reverseTheme true with reversible color", () => {
  const typography = render(
    <Typography reverseTheme={true} color="success">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with reverseTheme false with reversible color", () => {
  const typography = render(
    <Typography reverseTheme={false} color="success">
      Test Text
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with adjustsFontSizeToFit set to true", () => {
  const typography = render(
    <Typography adjustsFontSizeToFit={true} maxLines="single">
      Test Text that happens to be longer than the rest of the text. This just
      keeps going on and on. maxLines being set will make this work its magic
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text accessibilityRole", () => {
  const typography = render(
    <Typography accessibilityRole="text">Test Text</Typography>,
  );
  expect(typography.getByRole("text")).toBeDefined();
});

it("renders header accessibilityRole", () => {
  const typography = render(
    <Typography accessibilityRole="header">Test Text</Typography>,
  );
  expect(typography.getByRole("header")).toBeDefined();
});

it("renders text using the maxLines is also passed", () => {
  const typography = render(
    <Typography maxLines="small">
      Test Text that happens to be longer than the rest of the text. This just
      keeps going on and on. maxLines being set will make this work its magic
    </Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text with strikethough styling", () => {
  const typography = render(
    <Typography strikeThrough={true}>Test Text</Typography>,
  );
  expect(typography.toJSON()).toMatchSnapshot();
});

it("renders text that is inaccessible", () => {
  const typography = render(
    <Typography hideFromScreenReader={true}>Test Text</Typography>,
  );

  expect(typography.root.props).toEqual(
    expect.objectContaining({
      accessible: false,
      accessibilityRole: "none",
      importantForAccessibility: "no-hide-descendants",
    }),
  );
});
