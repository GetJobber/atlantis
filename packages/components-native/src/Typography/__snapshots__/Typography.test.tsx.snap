// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text respecting the text direction 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "right",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text using the maxLines is also passed 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  numberOfLines={2}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text that happens to be longer than the rest of the text. This just keeps going on and on. maxLines being set will make this work its magic
</Text>
`;

exports[`renders text with adjustsFontSizeToFit set to true 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={true}
  allowFontScaling={true}
  collapsable={false}
  numberOfLines={1}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text that happens to be longer than the rest of the text. This just keeps going on and on. maxLines being set will make this work its magic
</Text>
`;

exports[`renders text with black style and display as fontFamily 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "jobberpro-blk",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with bold style 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with bold style and display as fontFamily 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "jobberpro-bd",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with bold weight and italic style 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with center align 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "center",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with default color 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with default size 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with extraBold weight and display as fontFamily 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "jobberpro-xbd",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with green color 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgb(125, 176, 14)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with italic style 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with large size 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 18,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with letter spacing 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0.4,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with line height override 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 36,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with lowercase transform 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  test text
</Text>
`;

exports[`renders text with multiple properties 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-bold",
      },
      {
        "color": "rgba(255, 255, 255, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 18,
        "lineHeight": 22,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with no additional props 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with regular style 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with reverseTheme false with reversible color 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgb(81, 114, 9)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with reverseTheme true with reversible color 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgb(125, 176, 14)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with small size 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 14,
        "lineHeight": 18,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with strikethough styling 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
      {
        "textDecorationLine": "line-through",
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with uppercase transform 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(77, 105, 116, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  TEST TEXT
</Text>
`;

exports[`renders text with white color 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  selectable={true}
  selectionColor="rgb(132, 234, 0)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "rgba(255, 255, 255, 1)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;
